test/extern/ExternBlockClangError.good

```diff
@@ -1,9 +1,3 @@
-ExternBlockClangError.chpl:3:8: error: no previous prototype for function 'bar' [-Werror,-Wmissing-prototypes]
-  void bar(void) {
-       ^
-ExternBlockClangError.chpl:3:3: note: declare 'static' if the function is not intended to be used outside of this translation unit
-  void bar(void) {
-  ^
-  static 
```

Comment:
These code listings seem appropriate for `--detailed-errors`. Do you think it's possible to show them when we run that way? (Might be reasonable for a future-work issue; does not need to be addressed for this PR).

Reply:
Yes I think so, but not identical to how they previously looked. The `TextDiagnosticBuffer` gives us a `clang::SourceLocation` for each error, which lets us retrieve the line of C code it complained about. This wouldn't include carets or things like the `static` suggestion in this particular instance. We might be able to get more information if we used a different `DiagnosticConsumer` instead of a `TextDiagnosticBuffer`, or even wrote our own, but a lot of them seem to bring unwanted functionality with them.